cpp/dolfinx/fem/assemble_vector_impl.h

```diff
@@ -515,11 +558,11 @@ void assemble_cells(fem::DofTransformKernel<T> auto dof_transform,
     std::fill(be.begin(), be.end(), 0);
     kernel(be.data(), coeffs.data() + index * cstride, constants.data(),
            coordinate_dofs.data(), nullptr, nullptr);
-    dof_transform(_be, cell_info, c, 1);
+    dof_transform(_be, cell_info, c0, 1);
```

Comment:
Do we have a test that checks that with using `c` in the transform, that the code goes wrong?

Also, what mesh is cell_info associated with? In the codes above we use `cell_info0`, `cell_info1` explicitly, and I guess cell info for a submesh can be different than the parent mesh.

Reply:
We don't have a test for that, but it doesn't make sense to index the cell info from the test function mesh with `c`. 

`cell_info` is associated with the rest function mesh, I'll rename it to `cell_info0` for clarity.